docs/db/changelog/changesets/powerauth-java-server/1.9.x/20240718-add-temporary-keys.xml

```diff
@@ -0,0 +1,69 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ PowerAuth Server and related software components
+  ~ Copyright (C) 2024 Wultra s.r.o.
+  ~
+  ~ This program is free software: you can redistribute it and/or modify
+  ~ it under the terms of the GNU Affero General Public License as published
+  ~ by the Free Software Foundation, either version 3 of the License, or
+  ~ (at your option) any later version.
+  ~
+  ~ This program is distributed in the hope that it will be useful,
+  ~ but WITHOUT ANY WARRANTY; without even the implied warranty of
+  ~ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  ~ GNU Affero General Public License for more details.
+  ~
+  ~ You should have received a copy of the GNU Affero General Public License
+  ~ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+  -->
+
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.9.xsd">
+
+    <changeSet id="1" logicalFilePath="powerauth-java-server/1.9.x/20240718-add-temporary-keys.xml" author="Petr Dvorak">
+        <preConditions onFail="MARK_RAN">
+            <not>
+                <tableExists tableName="pa_temporary_key"/>
+            </not>
+        </preConditions>
+        <comment>Create a new table pa_temporary_key</comment>
+        <createTable tableName="pa_temporary_key">
+            <column name="id" type="varchar(37)">
+                <constraints primaryKey="true" />
+            </column>
+            <column name="application_key" type="varchar(32)">
+                <constraints nullable="false" foreignKeyName="pa_temporary_key_app_version_fk" referencedTableName="pa_application_version" referencedColumnNames="application_key"/>
```

Comment:
`pa_temporary_key_ application_key_fk`
this is to always easily detect which column is incorrectly inserted. You can have actually multiple FK from one table to another.

Reply:
In Oracle, max length of entity name is 30 characters.

Reply:
This is strange. The FK is on the column already. No need to put the column name to the FK name to duplicate it. You need to name it so you know what **target entity** it references.

Reply:
FYI on 19c Oracle (the minimum LTS) the limit for object names is 128 chars.

Reply:
Updated the names of FK